internal/bed/merging.go

```diff
@@ -0,0 +1,61 @@
+package bed
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+)
+
+// Merge lines in bed file
+func (bf *Bedfile) MergeLines() {
+	var merged Line
+	var mergedLines []Line
+	for i, l := range mergeSort(bf.Lines) {
+		// If the lines are overlapping or touching merge them
+		if i != 0 &&
+			merged.Chr == l.Chr &&
+			merged.Strand == l.Strand &&
+			merged.Feat == l.Feat &&
+			merged.Stop+bf.Overlap >= l.Start-1 {
+			// Set new stop if it is later than the
+			// merged stop
+			if l.Stop > merged.Stop {
+				merged.Stop = l.Stop
+				merged.Full[stopIdx] = strconv.Itoa(l.Stop)
+			}
+			// Join information in the optional columns
+			if len(l.Full) > stopIdx+1 {
+				for idx, col := range l.Full[stopIdx+1:] {
+					mIdx := idx + stopIdx + 1
+					if idxInSlice(strings.Split(merged.Full[mIdx], ","), col) == -1 {
```

Comment:
I would maybe make this its own function called something like `itemPresent(set string, item string) bool` and put the check in there. That might make it easier to read. Also, the `IdxInSlice()` function is only used here. Maybe they could be combined?

Reply:
I don't know if this is that much easier to read, but I I have now refactored `IdxInSlice` (now `stringInSlice` to return a bool instead of the index if present or -1 if not present.

Reply:
it is better now 👍